src/models/proof_abstractions/tasm/program.rs

```diff
@@ -657,4 +657,18 @@ pub mod test {
             .write_all(&proof_data)
             .expect("cannot write to file");
     }
+
+    #[macro_export]
+    macro_rules! catch_change_in_program {
+        ($consensus_program: expr, $hash_hex: literal) => {
```

Comment:
Nits only:

- I usually see no whitespace between the colon and the fragment specifier (“`expr`”, “`literal`”). Minor minor, though. 🤷
- I suggest to add `$(,)?` to allow an optional trailing comma.

```suggestion
        ($consensus_program:expr, $hash_hex:literal $(,)?) => {
```